pkg/monitors/jenkins/nodemetrics.go

```diff
@@ -0,0 +1,99 @@
+package jenkins
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	jc "github.com/signalfx/signalfx-agent/pkg/monitors/jenkins/client"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+)
+
+// map key:metric_name value:json_key
+var nodeMetricsMap = map[string]string{
+	"jenkins.node.vm.memory.total.used":    "vm.memory.total.used",
+	"jenkins.node.vm.memory.heap.usage":    "vm.memory.non-heap.used",
+	"jenkins.node.vm.memory.non-heap.used": "vm.memory.non-heap.used",
+	"jenkins.node.queue.size":              "jenkins.queue.size.value",
+	"jenkins.node.health-check.score":      "jenkins.health-check.score",
+	"jenkins.node.executor.count":          "jenkins.executor.count.value",
+	"jenkins.node.executor.in-use":         "jenkins.executor.in-use.value",
+}
+
+type NodeMetrics struct {
+	Version string                            `json:"version"`
+	Gauges  map[string]map[string]interface{} `json:"gauges"`
+}
+
+func nodeMetrics(jkClient jc.JenkinsClient, conf *Config) ([]*datapoint.Datapoint, error) {
```

Comment:
seems like this method does not really need the whole config, just the `metricsKey`. I think we should just pass the required field(s) from the config. same thing with other methods like `liveness`, `getUniqueDimension`.

Also, what do you think about adding methods like `nodeMetrics`, `liveness` etc on to a struct and initializing that struct in the `Configure` method, rather than passing the client around?

Reply:
As we discussed, made the change to only pass metricsKey 
As for the changing the functions into method with a receiver containing the jenkinsClient, with the last push of the code, it makes no sense to do that as each metric call will have its own jenkins client